279 weapon scaling

p-kaufmann opened the pull request:
I hope it is not too late (but still late, i know, im very sorry) to approve and merge into master (especially because this branch is rather big) and also that there aren't any large bugs in hiding.
Surely it will be fine and there are no bugs (clueless).

Weapons now scale indirectly with character level, through their stats. These increase their damage, attack frequency and in specific cases projectile explosion range (warhammer fireball), piercing (crossbow), bounces (lightningStaff) and range (dagger).
This might need a little bit more balancing, however i'd consider it done for this project in its current state.

Enemy scaling also got changed a little bit (increased health for bosses, enemy limit increase and reduced enemy movement speed scaling), Before bosses spawned on wave 2 and every 5th after. Now they still do but multiplied by the times this has happened (e.g wave 12 would be 3 bosses).

To implement the strength stat i had to change lots of code, even in server files, let me know if anything in the backend has broken. I basically searched the entire project for the speed stat and implemented the other character stats along side it.

Currently depending on weapon rng and heal rng you can completely AFK with 2 clients to wave 6-10, i hope this is fine because enemy speed (attack and movement) will scale until wave 41 is reached. Their health and damage should continue to scale endlessly.

It also would be great if, who ever looks over this can also tests this for 1-2 rounds and can give some feedback if changes are needed on the balancing side of things (weapon way too weak or too strong).

p-kaufmann committed:
Attempt to add MedicineBag as Weapon to BossShop
this is non-functional and will take up weaponslots even though no weapon is added

p-kaufmann committed:
Successfully added MedicineBag to BossShop
Fixed MedicineBag spam

Ben4566 committed:
Merge branch 'master' into 279-weapon-scaling

p-kaufmann committed:
Implemented rough weapon damage scaling
This scaling is not balanced and only scales damage

p-kaufmann committed:
Merge branch '279-weapon-scaling' of https://github.com/IP-2025/waves-of-the-fallen into 279-weapon-scaling

Commit:
Added shootDelay scaling to all weapons
Added Range scaling to Dagger
Added Fireball and HammerProjectile Radius scaling (without visual scaling)
Added Lightning additional chains scaling
Added Medicine slowdown scaling

None of these are tested for balance!!!

Commit:
Merge remote-tracking branch 'origin/master' into 279-weapon-scaling

Commit:
Added crossbow piercing scaling

Commit:
Fixed MagicStaffs cast time

Commit:
Merge remote-tracking branch 'origin/master' into 279-weapon-scaling

Commit:
edited 16!!! files to add strength :|
added all character stats to data transmission

Commit:
Swapped dummy stats with real weapon owner stats

Commit:
Merge remote-tracking branch 'origin/master' into 279-weapon-scaling

Commit:
Applied small tweaks to weapon scaling
Refactored some weapon scaling code
Medicine now scales with player values

Commit:
Changed enemy scaling, boss spawnlogic and base hp (UNTESTED)
Fixed healing of dead players

Commit:
Reduced max attack speed for Kunai

p-kaufmann committed:
AFK to wave 10 with healstaff and 2 maxed archers
Might need a little more balancing
Weapon damage should now be calculated correctly (this time for sure)
Slightly adjusted weapon damage and other scaling
Now works in local multiplayer too (hopefully not too unstable)

p-kaufmann committed:
Merge remote-tracking branch 'origin/master' into 279-weapon-scaling

p-kaufmann commented:
> i cant select a weapon in the popup menu

Bug also exists in master and has been there for a while, this was still functional in #423 branch

PizzaE4ter commented:
it seems there is a layer bug, because we cannot interact with the joystick or shop menu

p-kaufmann committed:
We do a little bit of balancing and bugfixing
Fixed shop hud being in background
Slightly adjusted boss hitbox
Slightly balanced healing/enemy damage
Slightly balanced dagger&doubleblades
Adjusted initial values of character_settings.cfg

p-kaufmann committed:
Merge remote-tracking branch 'origin/master' into 279-weapon-scaling
